MoppApp/MoppApp/ContainerActions.swift

```diff
@@ -121,21 +121,54 @@ extension ContainerActions where Self: UIViewController {
         let topSigningViewController = navController.viewControllers.last!
-        let containerViewController = topSigningViewController as? ContainerViewController
-        let containerPath = containerViewController!.containerPath
-        
-        MoppLibContainerActions.sharedInstance().addDataFilesToContainer(
-            withPath: containerPath,
-            withDataFilePaths: dataFilePaths,
-            success: { container in
-                landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: {
-                    containerViewController?.reloadContainer()
-                })
-            },
-            failure: { error in
-                landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: nil)
+
+        if (landingViewController.containerType == .asic) {
+            let containerViewController = topSigningViewController as? ContainerViewController
+            let containerPath = containerViewController!.containerPath
+            MoppLibContainerActions.sharedInstance().addDataFilesToContainer(
+                withPath: containerPath,
+                withDataFilePaths: dataFilePaths,
+                success: { container in
+                    landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: {
+                        containerViewController?.reloadContainer()
+                    })
+                },
+                failure: { error in
+                    landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: nil)
+                }
+            )
+        } else {
+            let containerViewController = topSigningViewController as? CryptoContainerViewController
+            dataFilePaths.forEach {
+                var filename = ($0 as NSString).lastPathComponent as NSString
+                filename = generateNewFilename(container :(containerViewController?.container)!, filename: filename, count: 0)
+                let dataFile = CryptoDataFile.init()
+                dataFile.filename = filename as String?
+                dataFile.filePath = $0
+                
+                containerViewController?.container.dataFiles.add(dataFile)
             }
-        )
+            
+            landingViewController.importProgressViewController.dismissRecursively(animated: false, completion: {
+                containerViewController?.reloadContainer()
+            })
+        }
+    }
+    
+    private func generateNewFilename(container: CryptoContainer, filename: NSString, count: Int) -> NSString {
+        var newFilename = filename
+        if (count > 0){
+            let fileExtension  = filename.pathExtension
+            let withoutExtension = filename.deletingPathExtension
+            newFilename = withoutExtension.appendingFormat("("+String(count)+")."+fileExtension) as NSString
```

Comment:
Swift's String Interpolation should be used whenever possible, for better code quality:
```swift
newFilename = withoutExtension.appendingFormat("\(String(count)).\(fileExtension)") as NSString
```